README.md

```diff
@@ -1,31 +1,654 @@
+encapsulate a collection of `Concert` class instances.
+The `ConcertRepository` class models a subset of
+functionality found in the `ArrayList`
+class that is part of the Java Collections Framework.
+
+
+(1) Edit the `ConcertRepository` class to add two instance variables:
+
+- An array named `concerts` that stores `Concert` class instances.
+- An int named `currentSize` that stores the number of `Concert` objects
+  that have been added to the array.  This value may differ from the array size,
+  which represents the maximum number of concerts that may be added to the array.
+
+
+(2) Create a constructor for the `ConcertRepository` that takes one parameter named
+`maxSize`.  The constructor should initialize the array size using the `maxSize` variable.
+Note that initially all elements in the array will be `null` since the array
+stores instances of the reference type  `Concert`.
+
+(3) Generate a getter method for the `currentSize` instance variable named `getCurrentSize`.
+
+(4) Create a method named `add` that takes one parameter and returns a `boolean`.
+The parameter type should be `Concert`.  
+
+- The `currentSize` instance variable represents the number of concerts that have
+  been added to the array. Initially this value is 0.
+- If the array is not full (i.e. `currentSize` is less than the array length),
+  add the parameter object into the array using `currentSize` as the index.  Increment
+  `currentSize` and return `true` to indicate the concert was added to the repository.
+- If the array is already full, simply return `false` to indicate the concert
+  could not be added to the repository.
+
+(5) Create a method named `get` that takes in int parameter and returns a `Concert` object.
+The method should return a `Concert` object using the parameter as the index
+into the array, or return `null` if the index is out of bounds.  The method should not
+throw an exception.
+
+(6) Edit the `ConcertRepositoryTest` Junit class to test the constructor and methods:
+
+```java
+
+class ConcertRepositoryTest {
+
+    @Test
+    void constructor() {
+        // repository can hold up to 3 concerts
+        ConcertRepository repository = new ConcertRepository(3);
+
+        // current size is 0 since no concerts have been added
+        assertEquals(0, repository.getCurrentSize());
+    }
+
+    @Test
+    void addGet1() {
+        // repository can hold up to 3 concerts
+        ConcertRepository repository = new ConcertRepository(3);
+        assertEquals(0, repository.getCurrentSize());
+
+        // add a concert
+        assertTrue(repository.add(new Concert("Artist0" , 1000)));
+        assertEquals(1, repository.getCurrentSize());
+
+        // retrieve the concert using index 0
+        Concert c = repository.get(0);
+        assertEquals("Artist0", c.getPerformer());
+        assertEquals(1000, c.getAvailable());
+        assertEquals(0, c.getPerformer());
+
+    }
+
+    @Test
+    void addget5() {
```

Comment:
Not very important: `get` should probably be capitalized

Reply:
Second - just for demonstration of camel casing in method names

Reply:
fixed